test/fixtures/test-runner/output/test-timeout-flag.js

```diff
@@ -0,0 +1,25 @@
+// Flags: --test-timeout=20
+'use strict';
+const { describe, test, afterEach } = require('node:test');
+const { setTimeout } = require('node:timers/promises');
+
+let timeout;
+
+describe('--test-timeout is set to 20ms', () => {
+  test('should timeout after 20ms', async () => {
+    timeout = await setTimeout(2000);
+  });
+  test('should timeout after 5ms', { timeout: 5 }, async () => {
+    timeout = await setTimeout(2000);
+  });
+  test('should not timeout', { timeout: 5000 }, async () => {
+    await setTimeout(200);
```

Comment:
I think this could be made much smaller without changing the semantics of the test. By making it as small as possible, the CI will run a bit faster.